src/gluonts/shell/README.md

```diff
@@ -154,3 +154,32 @@ as the environment-variable, instead of `DEEPAR_INFERENCE_CONFIG`.
    ...
 }
 ```
+
+## Dynamic Code
+
+It is possible to pass a `code` channel to training jobs to install code
+dependencies without the need to build a new container.
```

Comment:
As I understood it, you want to upload the directory/file `code` is pointing to to s3.
Would it make sense to run `python setup.py sdist` if the user input is a path to an installable package and upload the resulting `.tar.gz` instead of uploading the full package?
I know many times experiments data/results/scripts/log and much more are also saved n the package directory. It can get really expensive if at every minor change the whole directory is uploaded to s3.

Reply:
Yes, but that's on the client side. Here we need to consume whatever is given to us, which would include both zipped folders, and python source distribution.

On the client side (which is not part of this PR), we would do the `python setup.py sdist` thing to get a smaller archive :)